src/expr/skolem_manager.h

```diff
@@ -132,74 +132,187 @@ enum class SkolemFunId
    * i = 0, ..., n.
    */
   RE_UNFOLD_POS_COMPONENT,
-  BAGS_CARD_CARDINALITY,
+  /**
+   * An uninterpreted function for bag.fold operator:
+   * To compute (bag.card A), we need a function that
+   * counts multiplicities of distinct elements. We call this function
+   * combine of type Int -> Int where:
+   * combine(0) = 0.
+   * combine(i) = m(elements(i), A) + combine(i-1) for 1 <= i <= n.
+   * elements: a skolem function for (bag.fold f t A)
+  *            see BAGS_CARD_ELEMENTS.
+   * n: is the number of distinct elements in A.
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [0,n].
+   */
+  BAGS_CARD_COMBINE,
+  /**
+   * An uninterpreted function for bag.card operator:
+   * To compute (bag.card A), we need a function for
+   * distinct elements in A. We call this function
+   * elements of type Int -> T where T is the type of
+   * elements of A.
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [1,n].
+   */
   BAGS_CARD_ELEMENTS,
+  /**
+   * An uninterpreted function for bag.card operator:
+   * To compute (bag.card A), we need to guess n
+   * the number of distinct elements in A.
+   */
   BAGS_CARD_N,
+  /**
+   * An uninterpreted function for bag.card operator:
+   * To compute (bag.card A), we need a function for
+   * distinct elements in A which is given by elements defined in
+   * BAGS_CARD_ELEMENTS.
+   * We also need unionDisjoint: Int -> (Bag T) to compute
+   * the disjoint union such that:
+   * unionDisjoint(0) = bag.empty.
+   * unionDisjoint(i) = disjoint union of {<elements(i), m(elements(i), A)>}
+   * and unionDisjoint(i-1).
+   * unionDisjoint(n) = A.
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [0,n].
+   */
   BAGS_CARD_UNION_DISJOINT,
+  /**
+   * An uninterpreted function for bag.fold operator:
+   * To compute (bag.fold f t A), we need to guess the cardinality n of
+   * bag A using a skolem function with BAGS_FOLD_CARD id.
+   */
```

Comment:
Ditto, this should take the bag to fold as an argument, see https://github.com/cvc5/cvc5/blob/main/src/theory/bags/bag_reduction.cpp#L71C46-L71C60.